internal/state/relationship/capturer.go

```diff
@@ -95,8 +123,8 @@ func (c *Capturer) deleteForRoute(routeName types.NamespacedName) {
 	delete(c.routesToServices, routeName)
 }
 
-func (c *Capturer) decrementRefCount(svcName types.NamespacedName) {
-	if c.serviceRefCount[svcName] == 1 {
+func (c *CapturerImpl) decrementRefCount(svcName types.NamespacedName) {
+	if c.serviceRefCount[svcName] == 0 {
```

Comment:
because we remove only when it is 0 (not when 1 as it was before), can we have a case when a lot of no longer referenced services linger in the map? 

Ex:
a HTTPRoute that refs svcA is created  -> serviceRefCount is {'ns/A': 1}
the HTTPRoute is updated to ref svc B -> serviceRefCount is  {'ns/A': 0, 'ns/B': 1}

Will 'ns/A' ever get removed?

Reply:
I can update this to check if ref count is 0 || 1. That should handle both cases.

Reply:
https://github.com/nginxinc/nginx-kubernetes-gateway/pull/221/commits/59f8c711a25dedc95b14cf93c7a60dd25b1a758a